### ISSUE TITLE
SRI can't be determined with 0.5.1 release

### ISSUE THREAD

**Issue by noderat** [waysact/webpack-subresource-integrity#11]
After moving to 0.5.1 I'm now finding all of my chunks fail to generate a hash. (not using html-webpack-plugin)

Example:
```
WARNING in webpack-subresource-integrity: cannot determine hash for asset '../../../../../vendor.bundle.0e48a883951098cad7a7.js', the resource will be unprotected.
WARNING in webpack-subresource-integrity: cannot determine hash for asset '../../../../../app.bundle.0e48a883951098cad7a7.js', the resource will be unprotected.
```

**Comment by jscheid** [comment]
Hi, could you provide some more information? Ideal would be a test case or a sample repository demonstrating the bug, or at least a copy of your Webpack configuration.

**Comment by noderat** [comment]
I tried using html-webpack-plugin to see if that had any effect, though now I can tell it throws the warnings prior to even reaching the html plugin. Here's the webpack config from my latest attempt:
```javascript
const path = require('path');
const webpack = require('webpack');
const HtmlPlugin = require('html-webpack-plugin');
const CleanPlugin = require('clean-webpack-plugin');
const SriPlugin = require('webpack-subresource-integrity');

const config = require('./config');

const rootPath = path.resolve(__dirname);
const sourcePath = path.join(rootPath, './source');
const publicPath = path.join(rootPath, './public');

const DEV = (config.get('env') === 'development');

module.exports = {
  debug: DEV,
  context: path.join(rootPath, './source'),
  devtool: 'cheap-module-source-map',
  entry: {
    app: ['./app/app.js'],
  },
  output: {
    pathinfo: true,
    path: publicPath,
    publicPath: '/',
    filename: '[name].bundle.[hash].js',
    sourceMapFilename: '[file].map',
  },
  module: {
    preLoaders: [
      { test: /\.js$/, loader: 'source-map-loader' },
    ],
    loaders: [
      { test: /\.ejs$/, loader: 'ejs-compiled' },
      { test: /\.html$/, loader: 'html' },
      { test: /\.(scss|sass)$/, loaders: ['style', 'css?sourceMap', 'sass?sourceMap'] },
      { test: /\.css$/, loaders: ['style', 'css?sourceMap'] },
      { test: /\.svg$/, loader: 'file', include: path.join(sourcePath, './images/') },
      { test: /\.(jpg|png|gif)$/, loader: 'url?limit=25000', include: path.join(sourcePath, './images/') },
      {
        test: /\.jsx?$/,
        loaders: [
          'ng-annotate',
          `babel?${JSON.stringify({
            cacheDirectory: true,
          })}`,
        ],
        exclude: /node_modules/,
      },
      { test: /\.woff$/, loader: 'url?limit=65000&mimetype=application/font-woff&name=fonts/[name].[ext]' },
      { test: /\.woff2$/, loader: 'url?limit=65000&mimetype=application/font-woff2&name=fonts/[name].[ext]' },
      { test: /\.[ot]tf$/, loader: 'url?limit=65000&mimetype=application/octet-stream&name=fonts/[name].[ext]' },
    ],
  },
  plugins: [
    new CleanPlugin([publicPath], { root: rootPath, verbose: false }),
    new SriPlugin(['sha256', 'sha384']),
    new webpack.ProvidePlugin({
      path: 'path',
    }),
    new webpack.DefinePlugin({
      API_ENDPOINT: JSON.stringify(config.get('api')),
      DEVELOPMENT: (config.get('env') === 'development'),
      STAGING: (config.get('env') === 'staging'),
      PRODUCTION: (config.get('env') === 'production'),
    }),
    new webpack.optimize.CommonsChunkPlugin({
      name: 'vendor',
      minChunks(module) {
        return module.resource && module.resource.indexOf(sourcePath) === -1;
      },
    }),
    new HtmlPlugin({
      inject: false,
      template: path.join(sourcePath, './index.ejs'),
      hash: true,
      cache: true,
      mountPoint: '<app>Loading...</app>',
      minify: DEV ? undefined : {
        collapseWhitespace: true,
        conservativeCollapse: true,
        preserveLineBreaks: true,
        quoteCharacter: '"',
      },
    }),
  ],
  sassLoader: {
    data: `$env: ${config.get('env')};`,
  },
};
```

**Comment by jscheid** [comment]
Thanks, could you send over your `source/index.ejs` as well please?

**Comment by noderat** [comment]
Sure, though keep in mind this throws the same warnings even if webpack-html-plugin is not used. I've included both ways I've attempted to include SRI into the EJS template.

```ejs
<!DOCTYPE html>

<% var item, key %>

<% htmlWebpackPlugin.options.meta = htmlWebpackPlugin.options.meta || {} %>

<html lang="en">
<head>
  <meta charset="utf-8">
  <meta content="ie=edge" http-equiv="x-ua-compatible">

  <title ng-bind-template="{{ pageTitle ? pageTitle + ' - <%= htmlWebpackPlugin.options.title %>' : '<%= htmlWebpackPlugin.options.title %>' }}">
    <%= htmlWebpackPlugin.options.title %>
  </title>

  <% for (key in htmlWebpackPlugin.options.meta) { %>
    <meta content="<%= htmlWebpackPlugin.options.meta[key] %>" name="<%= key %>">
  <% } %>

  <% if (htmlWebpackPlugin.options.mobile) { %>
    <meta content="width=device-width, initial-scale=1" name="viewport">
  <% } %>

  <% if (htmlWebpackPlugin.files.favicon) { %>
    <link href="<%= htmlWebpackPlugin.files.favicon %>" rel="shortcut icon">
  <% } %>

  <% for (var key in htmlWebpackPlugin.files.css) { %>
    <link href="<%= htmlWebpackPlugin.files.css[key] %>" rel="stylesheet">
  <% } %>


  <% if (htmlWebpackPlugin.options.baseHref) { %>
    <base href="<%= htmlWebpackPlugin.options.baseHref %>" />
  <% } %>


</head>
<body>
  <% if (htmlWebpackPlugin.options.mountPoint) { %>
    <%- htmlWebpackPlugin.options.mountPoint %>
  <% } %>

  <% if (htmlWebpackPlugin.options.window) { %>
    <script type="text/javascript">
    <% for (key in htmlWebpackPlugin.options.window) { %>
      window['<%= key %>'] = <%= JSON.stringify(htmlWebpackPlugin.options.window[key]) %>;
    <% } %>
    </script>
  <% } %>

  <!-- This works on webpack SRI plugin 0.5.0 -->
  <% for (var chunk in htmlWebpackPlugin.files.chunks) { %>
    <script src="<%= htmlWebpackPlugin.files.chunks[chunk].entry %>"
      <% var basename = path.basename(htmlWebpackPlugin.files.chunks[chunk].entry);
         if (compilation.assets[basename] && compilation.assets[basename].integrity) { %>
            integrity="<%= compilation.assets[basename].integrity %>"
            crossorigin="anonymous"
      <% } %>
    ></script>
  <% } %>

  <!-- 
    This fails reguardless of webpack SRI plugin version and throws the following error:
    Cannot read property 'integrity' of undefined
  -->
  <% for (var chunk in htmlWebpackPlugin.files.chunks) { %>
    <% var src = htmlWebpackPlugin.files.chunks[chunk].entry; %>
    <script src="<%= src %>" integrity="<%= compilation.assets[src].integrity %>" crossorigin="anonymous"></script>
  <% } %>

  <% if (htmlWebpackPlugin.options.devServer) { %>
  <script src="<%= htmlWebpackPlugin.options.devServer%>/webpack-dev-server.js"></script>
  <% } %>

  <% if (htmlWebpackPlugin.options.googleAnalytics) { %>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    <% if (htmlWebpackPlugin.options.googleAnalytics.trackingId) { %>
      ga('create', '<%= htmlWebpackPlugin.options.googleAnalytics.trackingId%>', 'auto');
      <% } else { throw new Error("html-webpack-template requires googleAnalytics.trackingId config"); }%>
    <% if (htmlWebpackPlugin.options.googleAnalytics.pageViewOnLoad) { %>
      ga('send', 'pageview');
    <% } %>
  </script>
  <% } %>
</body>
</html>
```

**Comment by jscheid** [comment]
Thanks again for providing some context, it allowed me to reproduce and fix this. I'm planning to release the fix shortly as 0.5.2, however in the course of debugging this I found a potential issue with the documentation that I would like to investigate first, so it might take another day or two.

In the meantime, could you test the commit to make sure that it fixes your issue?

    npm install waysact/webpack-subresource-integrity#dc8a58

**Comment by noderat** [comment]
npm throws an error, perhaps it's something with your `package.json`?

```
webpack-subresource-integrity@0.5.1  invalid (git://github.com/waysact/webpack-subresource-integrity.git#dc8a58376661665fbba97f87364c8f543971b7e8)
```

**Comment by noderat** [comment]
It appears that even with the npm error, it correctly installed and appears to be working. I'm going to test a little more though since now the ejs template is failing with `Cannot read property 'integrity' of undefined` so I'm thinking the template code provided in your readme.md needs to check if the object exists first.

**Comment by jscheid** [comment]
@noderat that's the other unrelated issue I'm going to look at. (Unrelated in the sense that it doesn't have anything to do with the `... the resource will be unprotected` issue that you've reported.) I'm in the process of reworking that example code in the README, I'll let you know when I have something to look at. Maybe not this week as I'm going to be travelling. As a workaround you could enable `inject: true` if that's an option for you.

**Comment by noderat** [comment]
I'll try that for now, I wasn't planning on using webpack-html-plugin but it seems to fit my needs so I'll continue and see what I can come up with. Thanks for the assistance @jscheid, very much appreciated.

**Comment by noderat** [comment]
@jscheid Any update on getting a new release pushed with this fix?

**Comment by jscheid** [comment]
@noderat sorry it took so long, but this is released now in v0.6.0. See also the changes in #13 which you might find useful for when running with `inject: false`? I hope it all works for you now, let me know if something still isn't.

**Comment by noderat** [comment]
@jscheid Perfect! Great release!
